File: system-test/java/src/main/java/com/optivem/shop/dsl/driver/port/external/tax/TaxDriver.java
package com.optivem.shop.dsl.driver.port.external.tax;

import com.optivem.shop.dsl.driver.port.external.tax.dtos.GetCountryRequest;
import com.optivem.shop.dsl.driver.port.external.tax.dtos.GetCountryResponse;
import com.optivem.shop.dsl.driver.port.external.tax.dtos.ReturnsCountryRequest;
import com.optivem.shop.dsl.driver.port.shared.dtos.ErrorResponse;
import com.optivem.shop.dsl.common.Result;

public interface TaxDriver extends AutoCloseable {
    Result<Void, ErrorResponse> goToTax();

    Result<GetCountryResponse, ErrorResponse> getCountry(GetCountryRequest request);

    Result<Void, ErrorResponse> returnsCountry(ReturnsCountryRequest request);
}

File: system-test/typescript/src/testkit/driver/port/external/tax/tax-driver.ts
import type { Result } from '../../../../common/result.js';
import type { TaxErrorResponse } from './dtos/errors/TaxErrorResponse.js';
import type { GetCountryRequest } from './dtos/GetCountryRequest.js';
import type { GetTaxResponse } from './dtos/GetTaxResponse.js';
import type { ReturnsTaxRateRequest } from './dtos/ReturnsTaxRateRequest.js';

export interface TaxDriver {
  goToTax(): Promise<Result<void, TaxErrorResponse>>;
  getTaxRate(request: GetCountryRequest): Promise<Result<GetTaxResponse, TaxErrorResponse>>;
  returnsTaxRate(request: ReturnsTaxRateRequest): Promise<Result<void, TaxErrorResponse>>;
  close(): Promise<void>;
}
